src/suites/cts/validation/createBindGroupLayout.spec.ts

```diff
@@ -148,3 +148,91 @@ g.test('dynamic set to true is allowed only for buffers', async t => {
   { type: 'sampled-texture', _success: false },
   { type: 'storage-texture', _success: false },
 ]);
+
+g.test('number of resources per stage exceeds maximum value for resource type', async t => {
+  const { type, maximumCount } = t.params;
+
+  const maxResourceBindings: GPUBindGroupLayoutBinding[] = [];
+  for (let i = 0; i < maximumCount; i++) {
+    maxResourceBindings.push({
+      binding: i,
+      visibility: GPUShaderStage.FRAGMENT,
+      type
+    });
+  }
+
+  const goodDescriptor: GPUBindGroupLayoutDescriptor = { bindings: maxResourceBindings };
+
+  // Control
+  t.device.createBindGroupLayout(goodDescriptor);
+
+  const badDescriptor = clone(goodDescriptor);
+  badDescriptor.bindings.push({
+    binding: maximumCount,
+    visibility: GPUShaderStage.FRAGMENT,
+    type
+  });
+
+  t.expectValidationError(() => {
+    t.device.createBindGroupLayout(badDescriptor);
+  });
+}).params([
+  { type: 'uniform-buffer', maximumCount: 12 },
+  { type: 'storage-buffer', maximumCount: 4 },
+  { type: 'readonly-storage-buffer', maximumCount: 4 },
+  { type: 'sampler', maximumCount: 16 },
+  { type: 'sampled-texture', maximumCount: 16 },
+  { type: 'storage-texture', maximumCount: 4 },
+]);
```

Comment:
Another idea for a test, maybe later: create maximumCount of one type, and then add one more of a different type. In all cases except storage-buffer + readonly-storage-buffer, this should pass validation.